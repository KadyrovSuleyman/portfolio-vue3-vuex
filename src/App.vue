<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
});
</script>

<template>
  <main>Hello world</main>
</template>
