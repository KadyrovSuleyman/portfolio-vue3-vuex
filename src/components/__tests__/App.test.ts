import { mount, VueWrapper } from '@vue/test-utils';
import { createStore, Store } from 'vuex';
import App from '../App.vue';

jest.mock('../StakingAppBlock/state');
jest.mock('../StakingAppBlock/ActionsBlock/state.ts');
jest.mock('../StakingAppBlock/CalculatorBlock/state.ts');
jest.mock('../StakingAppBlock/StakeInfoBlock/InfoContainer/state.ts');
jest.mock('../StakingAppBlock/StakeInfoBlock/state.ts');
jest.mock('../StakingAppBlock/TariffsBlock/state.ts');

jest.mock('../Header/WalletBlock/WalletDiv/state.ts');
jest.mock('../Header/WalletBlock/state.ts');

jest.mock('../Modals/ConnectWalletModal/WalletsList/state.ts');
jest.mock('../Modals/ConnectWalletModal/state.ts');
jest.mock('../Modals/Copied/state.ts');
jest.mock('../Modals/ReplenishModal/Calculator/state.ts');
jest.mock('../Modals/ReplenishModal/state.ts');
jest.mock('../Modals/TransactionConfirmed/state.ts');

let store: Store<any>;
beforeEach(() => {
  store = createStore<any>({});
});

beforeEach(() => {
  const el = document.createElement('div');
  el.className = 'app';
  document.body.appendChild(el);
});
afterEach(() => {
  document.body.outerHTML = '';
});

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('app renders', () => {
  wrapper = mount(App, {
    global: { plugins: [store] },
  });
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  wrapper = mount(App, {
    global: { plugins: [store] },
  });
  expect(wrapper.find('.app').classes()).toEqual(['app']);

  wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: true,
    },
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.app').classes()).toEqual(['app', 'app__selected']);

  wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: false,
    },
  });
  await wrapper.vm.$nextTick();
  expect(wrapper.find('.app').classes()).toEqual(['app']);
});
