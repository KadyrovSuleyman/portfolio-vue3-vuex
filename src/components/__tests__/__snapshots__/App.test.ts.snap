// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders 1`] = `
<div data-v-app="">
  <div class="app">
    <header class="app-header">
      <div class="header-navPanel">
        <div class="navPanel-navButton navPanel-navButton__selected"><a href="/stacking" class="navButton-link navButton-link__selected">Stacking</a>
          <div class="navButton-rectangle navButton-rectangle__selected"></div>
        </div>
        <div class="navPanel-navButton"><a href="/bridge" class="navButton-link">Bridge</a>
          <div class="navButton-rectangle"></div>
        </div>
        <div class="navPanel-navButton"><a href="/sho" class="navButton-link">SHO</a>
          <div class="navButton-rectangle"></div>
        </div>
      </div>
      <div class="header-walletBlock">
        <div class="walletBlock-connectBlock"><button class="connectBlock-button"> Connect wallet </button></div>
      </div>
    </header>
    <div class="app-headerBorder"></div>
    <div class="app-stakingAppBlock"><span class="stakingAppBlock-header"> Staking App </span>
      <!--v-if-->
      <div class="stakingAppBlock-tariffsBlock"></div>
      <div class="stakingAppBlock-calculatorBlock">
        <div class="calculatorBlock-infoBlock">
          <div class="infoBlock-div infoBlock-div__icon_warning"></div><span class="infoBlock-span"> To perform actions on the page, connect your wallet </span>
        </div>
      </div>
      <div class="stakingAppBlock-actionsBlock">
        <!--v-if--><button class="actionsBlock-mainButton">
          <!--v-if--><span class="mainButton-span">Connect wallet</span>
        </button>
        <!--v-if-->
        <!--v-if-->
      </div>
      <!--v-if--><button class="stakingAppBlock-viewContractButton"><span class="viewContractButton-span"> View contract </span></button>
    </div>
    <div class="app-footer"></div>
  </div>
  <!--teleport start-->
  <!--teleport end-->
  <!--teleport start-->
  <!--teleport end-->
  <!--teleport start-->
  <!--teleport end-->
  <!--teleport start-->
  <!--teleport end-->
</div>
`;
